sdk/samples/iot/hub/src/sample_pnp_thermostat_component.c

```diff
@@ -0,0 +1,363 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+#include <stdint.h>
+#include <stdio.h>
+#include <time.h>
+
+#include <azure/core/az_json.h>
+#include <azure/core/az_result.h>
+#include <azure/core/az_span.h>
+
+#include "pnp_helper.h"
+#include "sample_pnp_thermostat_component.h"
+
+#ifdef _MSC_VER
+// "'getenv': This function or variable may be unsafe. Consider using _dupenv_s instead."
+#pragma warning(disable : 4996)
+#endif
+
+#define DOUBLE_DECIMAL_PLACE_DIGITS 2
+
+// IoT Telemetry Values
+static const az_span telemetry_name = AZ_SPAN_LITERAL_FROM_STR("temperature");
+
+// IoT Hub Commands Values
+static const az_span report_command_name_span = AZ_SPAN_LITERAL_FROM_STR("getMaxMinReport");
+static const az_span report_max_temp_name_span = AZ_SPAN_LITERAL_FROM_STR("maxTemp");
+static const az_span report_min_temp_name_span = AZ_SPAN_LITERAL_FROM_STR("minTemp");
+static const az_span report_avg_temp_name_span = AZ_SPAN_LITERAL_FROM_STR("avgTemp");
+static const az_span report_start_time_name_span = AZ_SPAN_LITERAL_FROM_STR("startTime");
+static const az_span report_end_time_name_span = AZ_SPAN_LITERAL_FROM_STR("endTime");
+static const az_span report_error_payload = AZ_SPAN_LITERAL_FROM_STR("{}");
+static char end_time_buffer[32];
+static char incoming_since_value[32];
+
+// Twin Values
+static const az_span desired_temp_property_name = AZ_SPAN_LITERAL_FROM_STR("targetTemperature");
```

Comment:
why do you sometimes add _span at the end like above, and something not?

Reply:
removed the `_span`